src/core/RandomVariables.jl

```diff
@@ -656,21 +659,26 @@ Symbol(vn::VarName) = Symbol(string(vn, all=false))  # simplified symbol
 
 Returns `true` if `vn`'s symbol is in `space` and `false` otherwise.
 """
-function in(vn::VarName, space::Set)::Bool
+function in(vn::VarName, space::Tuple)::Bool
     if vn.sym in space
         return true
     else
-        # Collect expressions from space
-        exprs = filter(el -> isa(el, Expr), space)
-        # Filter `(` and `)` out and get a string representation of `exprs`
-        expr_strs = Set((replace(string(ex), r"\(|\)" => "") for ex in exprs))
         # String representation of `vn`
         vn_str = string(vn, all=false)
-        # Check if `vn_str` is in `expr_strs`
-        valid = filter(str -> occursin(str, vn_str), expr_strs)
-        return length(valid) > 0
+        return _in(vn_str, space)
     end
 end
+@inline function _in(vn_str::String, space::Tuple)
```

Comment:
Not sure how. Most of the time this function won't be called anyways. I think it only gets called when `space` is somewhat funny, not just symbols. Not sure who wrote this but I don't fully understand when it is triggered. From our tests, mh.jl file, I believe the following alg uses this part of the code: 
```
gibbs = Gibbs(1000, CSMC(10, 1, :z1, :z2, :z3, :z4), MH(10, (:mu1,GKernel(1)), (:mu2,GKernel(1))))
```
I can come back to this later since it is a niche use case. If you have an improvement in mind, please open an issue.